app/poros/neo_search.rb

```diff
@@ -8,33 +8,42 @@ def birthday_data(birthdate)
     NeoService.new.birthday(birthdate)
   end
 
+  def daily_message
+    unless DailyMessage.where(date: Date.today.to_s).empty?
+      return @message = DailyMessage.where(date: Date.today.to_s).first
+    else
+      daily_neos
+    end
+  end
+
+  def daily_neos
+    message = new_daily_message
+    names = Name.new
+    events = NeoService.new.daily
+      events.map do |neo|
+      harbinger = message.harbingers.create(harbinger_params(neo, names.get_name))
+      harbinger.events.create(event_params(neo))
+    end
+    message
+  end
```

Comment:
this brings in names from the names list now and ties the daily message to harbingers with corresponding events.